auth/entity/memory.go

```
@@ -26,14 +28,57 @@ func (c *MemoryApiClient) GetApiKey() string {
 }
 
+func (c *MemoryApiClient) GetCurrentApiKey() contract.ApiClientKeyInterface {
+	return c.CurrentApiKey
+}
+
+func (c *MemoryApiClient) SetCurrentApiKey(key contract.ApiClientKeyInterface) {
+	c.CurrentApiKey = key.(*MemoryApiClientKey)
+}
+
 func (c *MemoryApiClient) GetClientScope() *contract.AccessScope {
+	if c.GetCurrentApiKey() != nil {
```

golangci-lint (1.62.2)

SA4023: this comparison is always true

(staticcheck)